packages/terra-notification-dialog/src/_transientUtils.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { createContext } from 'react';
+import { render, unmountComponentAtNode } from 'react-dom';
+import Base from 'terra-base';
+
+const TransientComponentContext = createContext();
+
+/**
+ * Mounts a transient component to the document. The component will be wrapped by a Base component to provide i18n.
+ * @param {Component} component The component instance to be mounted.
+ * @param {String} containerId An id that will be used to create the element to which the component will be mounted.
+ * @param {String} locale The string locale used to load translated messages.
+ * @param {Object} customMessages The object containing custom messages for the given locale.
+ * @returns {Promise} A Promise is returned. If a component is already mounted for the given id value, the Promise will
+ *                    be rejected. Otherwise, the Promise will be resolved after the component has been mounted.
+ */
+const mount = (component, containerId, locale, customMessages) => {
+  /**
+   * If an element with the chosen id is already present, the new presentation is aborted.
+   */
+  if (document.getElementById(containerId)) {
+    return Promise.reject();
+  }
+
+  const mountElement = document.createElement('div');
+  mountElement.id = containerId;
+  mountElement.setAttribute('data-terra-transient-component', true);
```

Comment:
I added an attribute here in case we ever needed to forcefully find and remove all transient components (in the case of session logouts or something). There's a lot more thought that needs to go into that.